model/model.go

```diff
@@ -317,12 +340,23 @@ func build(kubeClient *kubernetes.Clientset, routerDeployment *v1beta1ext.Deploy
 	return routerConfig, nil
 }
 
-func buildRouterConfig(routerDeployment *v1beta1.Deployment, platformCertSecret *v1.Secret, dhParamSecret *v1.Secret) (*RouterConfig, error) {
+func buildRouterConfig(routerDeployment *v1beta1.Deployment, configMap *v1.ConfigMap, platformCertSecret *v1.Secret, dhParamSecret *v1.Secret) (*RouterConfig, error) {
 	routerConfig := newRouterConfig()
-	err := modeler.MapToModel(routerDeployment.Annotations, "nginx", routerConfig)
+
+	if configMap.Data != nil {
+		err := configmapModeler.MapToModel(configMap.Data, "nginx", routerConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// To maintain backwards compatibility with previous Deis releases and Kube < 1.2
+	// annotations on the router Deployment override the ConfigMap.
+	err := annotationModeler.MapToModel(routerDeployment.Annotations, "nginx", routerConfig)
```

Comment:
In the end, I really don't care. I took this on because of the multiple mentions of wanting to transition to using configmaps over annotations. If you've decided you don't like it after all, I'll just close this PR and focus on implementing some test coverage for the annotation configuration method.